website/search/elastic_search.py

```diff
@@ -212,23 +213,31 @@ def load_parent(parent_id):
     return parent_info
 
 
+def get_doctype_from_node(node):
+    component_categories = [k for k in Node.CATEGORY_MAP.keys() if not k == 'project']
```

Comment:
This only needs to be computed once. Maybe make this a constant outside this function. Also, we can make this a set for more efficient `in` checks.